src/vaults/VaultRouter.sol

```diff
@@ -236,37 +236,6 @@ contract VaultRouter is Auth, Multicall, Recoverable, IVaultRouter {
         IAsyncVault(vault).claimCancelRedeemRequest(REQUEST_ID, receiver, controller);
     }
 
-    // --- Transfer ---
-    /// @inheritdoc IVaultRouter
-    function transferShares(uint16 centrifugeId, address vault, bytes32 receiver, uint128 amount)
-        public
-        payable
-        protected
-    {
-        SafeTransferLib.safeTransferFrom(IAsyncVault(vault).share(), msg.sender, address(this), amount);
-        _approveMax(IAsyncVault(vault).share(), 0, address(poolManager));
-        _pay();
-        IPoolManager(poolManager).transferShares(
-            centrifugeId, IAsyncVault(vault).poolId(), IAsyncVault(vault).trancheId(), receiver, amount
-        );
-    }
-
-    /// @inheritdoc IVaultRouter
-    function transferShares(uint16 centrifugeId, address vault, address receiver, uint128 amount)
-        external
-        payable
-        protected
-    {
-        transferShares(centrifugeId, vault, receiver.toBytes32(), amount);
-    }
-
-    // --- Register Asset ---
-    /// @inheritdoc IVaultRouter
-    function registerAsset(uint16 centrifugeId, address asset, uint256 tokenId) public payable {
```

Comment:
These messages can no longer be batched using `VaultRouter.multicall()`, are we ok with this? I think could be use cases where you would want to transfer shares to different locations at once.

Reply:
Yeah I think that is fine. It is a tradeoff but I just don't see a better option.

Reply:
What was the problem with forcing them to be used in the `VaultRouter`?

Reply:
The `PoolManager` needs to be a ward to call `authTransferFrom`. By doing this, no ERC20 approval is needed.